### DISCUSSION TITLE
Ruby problem while installing Homebrew

### DISCUSSION THREAD

**Discussion by EddFreeman** [Homebrew/discussions#3317] [Getting started] [answered]
I am trying to install Homebrew on my Windows 10 desktop running Ubuntu 22.04 installed into the Windows Subsystem for Linux (WSL). WSL and Ubuntu were installed yesterday, and this install of Homebrew is the first thing I am trying to do. The install ran for some minutes, then failed with the following messages.

==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:cd7fffb18ef9338baa670fc5e8fce99b0e3cc0f0fd7627bcbb56f3c8d54161d4
######################################################################## 100.0%
==> Pouring portable-ruby-2.6.8.x86_64_linux.bottle.tar.gz
/usr/bin/gzip: 1: ELF    : not found
/usr/bin/gzip: 3:    : not found
/usr/bin/gzip: 4: Syntax error: "(" unexpected
tar: Child returned status 2
tar: Error is not recoverable: exiting now
Error: Failed to install ruby 2.6.8!
Error: Failed to install Homebrew Portable Ruby and cannot find another Ruby 2.6.8!
If there's no Homebrew Portable Ruby available for your processor:
- install Ruby 2.6.8 with your system package manager (or rbenv/ruby-build)
- make it first in your PATH
- try again

Failed during: /home/linuxbrew/.linuxbrew/bin/brew update --force –quiet

I tried running the suggestion above rbenv/ ruby-build but got No such file or directory. I used to use Unix quite a bit, but it was 20 years ago and my memory isn’t what is was, can anyone please suggest the best way to resolve this Ruby issue? Oh, I also looked at the Ubuntu path, it appears to have references to all the programs and applications in my Windows 10 installation. Perhaps this is the error, possibly due to WSL?

Thanks,

Richard

**Comment by Bo98** [comment, marked as answer]
Are you using WSL1? WSL1 does have some issues. Do you need to use WSL1 or can you switch to WSL2?

**Reply by EddFreeman** [reply]
Thanks for your prompt reply. 
Google says that if the windows build is less than 18917 you have WSL1. My build is 19044. But then I found the suggestion to enter wsl -l -v it reported Ubuntu-22.04 version 1.
So I guess that it is WSL1 that I have. I didn't notice a choice when I installed it from the Microsoft Store, but I'll uninstall my current version and see if I can find WSL2 somewhere. I'll report back later on how I get on.

**Reply by CRagoGenesys** [reply]
I encountered the same problem but it worked fine after uninstalling Ubuntu LTS 20.04 on WSL1, enabled WSL2 and reinstalling Ubuntu LTS 20.04.

Followed this linked to enable WSL2

https://www.altisconsulting.com/au/insights/installing-ubuntu-bash-for-windows-10-wsl2-setup/

Used this link to install WSL2 Kernel
https://github.com/MicrosoftDocs/WSL/issues/977

Then reinstall Ubuntu LTS via Microsoft Store.

**Comment by EddFreeman** [comment]
After quite a bit of research it appears that if you install WSL on a computer that has the Virtual Machine Platform enabled then you will find you have WSL2. Otherwise you get WSL1. I installed the VMP, but then found that I couldn't open my Virtual Box virtual computer session. Further research suggested that this has also affected other users. As I need my Vbox system for a legacy program that won't run on new windows versions I have decided to forgo the Homebrew installation. I may revisit this sometime in the future after upgrades.

**Reply by Bo98** [reply]
It seems to be a bit random whether WSL1 works or not. I'm aware it works fine for some people, but for others not at all. I have no idea why, but it's likely a WSL1 bug unfortunately.

**Comment by CRagoGenesys** [comment]
Getting the same error as well on newly installed Ubuntu WSL 20.04 on windows 10, I was able to install this in the past probably last may not sure why it is not working now.

cjadmin@MNL-LOAN16-L:~$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
==> This script will install:
/home/linuxbrew/.linuxbrew/bin/brew
/home/linuxbrew/.linuxbrew/share/doc/homebrew
/home/linuxbrew/.linuxbrew/share/man/man1/brew.1
/home/linuxbrew/.linuxbrew/share/zsh/site-functions/_brew
/home/linuxbrew/.linuxbrew/etc/bash_completion.d/brew
/home/linuxbrew/.linuxbrew/Homebrew

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /bin/chown -R cjadmin:cjadmin /home/linuxbrew/.linuxbrew/Homebrew
==> Downloading and installing Homebrew...
Updating files: 100% (2801/2801), done.
HEAD is now at 5744895f3 Merge pull request #13490 from Homebrew/dependabot/bundler/Library/Homebrew/sorbet-static-and-runtime-0.5.10132
To restore the stashed changes to /home/linuxbrew/.linuxbrew/Homebrew, run:
  cd /home/linuxbrew/.linuxbrew/Homebrew && git stash pop
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:fc45ee6eddf4c7a17f4373dde7b1bc8a58255ea61e6847d3bf895225b28d072a
Already downloaded: /home/cjadmin/.cache/Homebrew/portable-ruby-2.6.8_1.x86_64_linux.bottle.tar.gz
==> Pouring portable-ruby-2.6.8_1.x86_64_linux.bottle.tar.gz
/usr/bin/gzip: 1: ELF: not found
/usr/bin/gzip: 3: : not found
/usr/bin/gzip: 4: Syntax error: "(" unexpected
tar: Child returned status 2
tar: Error is not recoverable: exiting now
Error: Failed to install ruby 2.6.8_1!
Error: Failed to install Homebrew Portable Ruby and cannot find another Ruby 2.6.8!
If there's no Homebrew Portable Ruby available for your processor:
- install Ruby 2.6.8 with your system package manager (or rbenv/ruby-build)
- make it first in your PATH
- try again

Failed during: /home/linuxbrew/.linuxbrew/bin/brew update --force --quiet


I already added the following different sub versions of Ruby 2.6.8 and also installed ruby via package manager (rbenv) but still does not fix my issue

cjadmin@MNL-LOAN16-L:~$ cd /home/cjadmin/.cache/Homebrew/
cjadmin@MNL-LOAN16-L:~/.cache/Homebrew$ ls -lah
total 31M
drwxr-xr-x 1 cjadmin cjadmin 512 Jun 30 15:21 .
drwxr-xr-x 1 cjadmin cjadmin 512 Jun 30 13:20 ..
-rw-r--r-- 1 cjadmin cjadmin   0 Jun 30 16:16 .cleaned
-rw-r--r-- 1 cjadmin cjadmin 11M Dec  8  2021 portable-ruby-2.6.3_2.x86_64_linux.bottle.tar.gz
-r-xr-xr-x 1 cjadmin cjadmin 11M Jun 30 15:21 portable-ruby-2.6.8.x86_64_linux.bottle.tar.gz
-rw-r--r-- 1 cjadmin cjadmin 11M Jun  4 00:56 portable-ruby-2.6.8_1.x86_64_linux.bottle.tar.gz

From the comment made by [Bo98] the answer would be to change to wsl2 unfortunately the ubuntu wsl available for windows in only wsl 1

PS C:\Users\crago> wsl -l -v
  NAME            STATE           VERSION
* Ubuntu-22.04    Running         1

**Reply by gromgit** [reply]
Did you read the previous threads in this discussion? Specifically https://github.com/orgs/Homebrew/discussions/3317#discussioncomment-2852117.

WSL1 is, to coin a phrase, **Not Real Linux**. Try installing WSL2, then reinstalling Homebrew.
